core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -158,15 +156,20 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
   }
 
   def asProjectOwner(projectName: RawlsBillingProjectName)(op: => Future[PerRequestMessage]): Future[PerRequestMessage] = {
-    val isOwner = dataSource.inTransaction { dataAccess =>
-      dataAccess.rawlsBillingProjectQuery.hasOneOfProjectRole(projectName, RawlsUser(userInfo), Set(ProjectRoles.Owner))
-    }
-    isOwner flatMap {
+    hasOneOfProjectRole(projectName, Set(ProjectRoles.Owner), userInfo) flatMap {
       case true => op
       case false => Future.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Forbidden, "You must be a project owner.")))
     }
   }
 
+  private def hasOneOfProjectRole(projectName: RawlsBillingProjectName, roles: Set[ProjectRoles.ProjectRole], userInfo: UserInfo): Future[Boolean] = {
+    samDAO.getResourcePolicies(SamResourceTypeNames.billingProject, projectName.value, userInfo).map { policies =>
+      val roles = policies.map(_.policy).flatMap(_.roles)
+
+      roles.contains(ProjectRoles.Owner.toString)
```

Comment:
something isn't right here, locally defined roles is shadowing the parameter named roles.

also we should not be checking roles anywhere, we should be checking actions